### PR TITLE
make example image package selections reusable

### PR DESCRIPTION
**Pull request by fmoessbauer**
This patch carves out the package selections from the example image. The selections provide great value for downstream users that do not want to base their image on the example image, but still want to avoid re-creating these selections to e.g. make wifi work.

Signed-off-by: Felix Moessbauer <felix.moessbauer@siemens.com>

### CODE REVIEW

**Comment by fmoessbauer** [comment]
@BaochengSu : I'll keep you as the author of the package-selections file, as my only contribution was to copy the content.

**Comment by jan-kiszka** [comment]
I vaguely recall a proposal by @AsuraZeng to have a more fine-grained feature selection, also via Kconfig. Don't find this anymore, but this here should be aligned while we are refactoring.

**Comment by AsuraZeng** [comment]
> I vaguely recall a proposal by @AsuraZeng to have a more fine-grained feature selection, also via Kconfig. Don't find this anymore, but this here should be aligned while we are refactoring.


yes, my proposal at the time was to exclude the necessary software, divide other software according to function blocks and those could be selected through kconfig.

**Comment by fmoessbauer** [comment]
Independent of the feature-selection concept, I still believe that this patch is valuable. Especially for downstream users it is much easier to use the provided variables directly. Also things like inspection and removal from these vars is possible, which is hard to achieve with a pure feature-selection based approach.

In short, we don't want to use the example image, but still do want to use pre-selected packages for the wifi.

**Comment by jan-kiszka** [comment]
If there is no overlap or immediate conflict with the feature selection (as I said, I didn't find the changes anymore), then I'm fine with this step.

File: recipes-core/images/iot2050-package-selections.inc [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    libstdc++6:armhf \
+    "
+
+# docker support
+IOT2050_DEBIAN_DOCKER_PACKAGES = " \
+    docker.io \
+    docker-compose \
+    "
+# coral support
+IOT2050_CORAL_PACKAGES = " \
```
Comment (AsuraZeng) [inline recipes-core/images/iot2050-package-selections.inc:99]: for coral support is from meta-coral. others are from debian repo. Can optimize the granularity of division and distinguish the source?

Comment (fmoessbauer) [reply recipes-core/images/iot2050-package-selections.inc:99]: @AsuraZeng : Yes, that's true. But in the end the package selection is also curated by the meta-iot2050 project. By that, IMHO it makes more sense to name the variable `IOT2050_CORAL_PACKAGES`.

**Comment by fmoessbauer** [comment]
@AsuraZeng any news on this topic? Is there something missing from my side?

**Review by AsuraZeng — APPROVED** [review]

**Pull request merged**